### PR TITLE
Create Header: Create projects button and dropdown 

### PR DESCRIPTION
**Pull request by Erin007**
Continuing the work outlined in #24917. This PR includes the create button and dropdown for different project types.

If `request.cookies['pm'] == 'create_header_2018'`, you'll now see: 

Dashboard: 
screen shot 2018-09-20 at 3 51 00 pm

Pegasus:
screen shot 2018-09-20 at 5 49 14 pm

Without the pm cookie set, there should be no user impact; everyone will see see the regular header. 

 Additional follow up needed not addressed in this first pass: 
- [x] This button should not show when on a level or project page
- [x] App Lab and Game Lab should be swapped for other project types if we know the student is younger than 13 
- [x] This button should only show on screens >=1024px wide
- [x] With an eventual 4 dropdowns in the header, I need to ensure that only one is open at a time 
- [x] Other layering bugs, tracked [here](https://docs.google.com/document/d/1aN0iDj4HCAcTIN8QI8OlkBfhsQFzXtpzR9P-0cR_5jM/edit)

### CODE REVIEW

File: dashboard/app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -350,28 +350,74 @@ img.video_thumbnail {
     }
 
     .user_menu_arrow_down,
-    .user_menu_arrow_up {
+    .user_menu_arrow_up,
+    .create_menu_arrow_down,
+    .create_menu_arrow_up, {
```
Comment (breville) [inline dashboard/app/assets/stylesheets/application.scss:355]: We could also create a common class for these, say `.create_menu_arrow`.

File: dashboard/app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -350,28 +350,74 @@ img.video_thumbnail {
...
       font-size: 30px;
     }
 
-    .user_options {
+    .user_options, {
```
Comment (breville) [inline dashboard/app/assets/stylesheets/application.scss]: nit: unnecessary comma?

File: dashboard/app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -350,28 +350,74 @@ img.video_thumbnail {
...
+        height: 70px;
+        width: 198px;
+        float: left;
+        border-left: 1px solid $charcoal;
+        border-right: 1px solid $charcoal;
+        border-top: 1px solid $charcoal;
+      }
+      a:hover {
+        color: $teal;
+        font-weight: 900;
```
Comment (breville) [inline dashboard/app/assets/stylesheets/application.scss]: Does this work?  I thought our use of web fonts meant that we usually chose a different font face if we wanted a different weight.

File: shared/css/header.scss [hunk 4]
```diff
@@ -155,6 +157,46 @@
     border-width: 0px;
   }
 
+  .create_options {
```
Comment (breville) [inline shared/css/header.scss:160]: If these are the same as above we could move them out into common CSS shared by both dashboard & pegasus?

File: shared/haml/user_header.haml [hunk 5]
```diff
@@ -2,6 +2,32 @@
...
+        = I18n.t("nav.create_button.create")
+      &nbsp;
+      %i.create_menu_arrow_down{class: "fa fa-caret-down"}
+      %i.create_menu_arrow_up{class: "fa fa-caret-up", style: "display: none"}
+      .create_options{style: 'display: none'}
+        %a.project_link_box{href: CDO.studio_url('projects/playlab/new')}
+          %img{src: "/shared/images/fill-70x70/courses/logo_playlab.png"}
+          .project_link
+            = I18n.t("nav.create_button.playlab")
+        %a.project_link_box{href: CDO.studio_url('projects/artist/new')}
```
Comment (breville) [inline shared/haml/user_header.haml:16]: Can we write some Ruby to generate these repeated options?

File: shared/haml/user_header.haml [hunk 6]
```diff
@@ -120,6 +146,31 @@
       });
     });
 
+    function hideCreateOptions() {
```
Comment (breville) [inline shared/haml/user_header.haml:149]: We should move this to a standalone .js file, even if we are going to inline it for performance reasons.

**Comment by breville** [comment]
Aesthetically, it seems that we align the right side of the menu with the right side of the menu bar icon for the other two dropdown, so should we do the same with this one?

File: shared/css/header.scss (hunk 4 above)
Comment (breville) [reply shared/css/header.scss:160]: See https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/hamburger/hamburger.js and https://github.com/code-dot-org/code-dot-org/blob/staging/shared/css/hamburger.scss for examples of common CSS and JS for the hamburger.

File: dashboard/app/assets/stylesheets/application.scss (hunk 1 above)
Comment (Erin007) [reply dashboard/app/assets/stylesheets/application.scss:355]: I use the down and up versions for separate styling later in the file and these classes are consistent with user_menu.

**Comment by Erin007** [comment]
Since this is all still behind a flag and I've done some branching gymnastics, I'm going to merge what I have so far then do a second pass/refactor to incorporate feedback around shared css, shared js, and using ruby to generate options.

File: shared/haml/user_header.haml (hunk 5 above)
Comment (Erin007) [reply shared/haml/user_header.haml:16]: Yes! Followed up in #24994

**Pull request merged**